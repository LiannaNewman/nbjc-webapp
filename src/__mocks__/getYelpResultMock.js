const getYelpResultMock = (overrides = {}) => {
  const business = {
    id: '123',
    name: 'La colombe coffee roasters',
    category: 'Coffee',
    address: '924 Blagden Alley Way Washington, D.C. 20001',
    phoneNumber: '(123) 456-7890',
  };
  return {
    ...business,
    ...overrides,
  };
};

export default getYelpResultMock;
