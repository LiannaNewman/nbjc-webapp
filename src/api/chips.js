export default [{
  name: 'Accessible',
  value: '1',
}, {
  name: 'ATM Available',
  value: '2',
}, {
  name: 'ASL Spoken',
  value: '3',
}, {
  name: 'Black Friendly',
  value: '4',
}, {
  name: 'Black Owned',
  value: '5',
}, {
  name: 'Gender-Neutral Restroom',
  value: '6',
}, {
  name: 'Inclusive',
  value: '7',
}, {
  name: 'LGBTQIA+ Friendly',
  value: '8',
}, {
  name: 'Multilingual',
  value: '9',
}, {
  name: 'Person of Color owned',
  value: '10',
}, {
  name: 'Price Conscious',
  value: '11',
}, {
  name: 'Trans-Trained',
  value: '12',
}, {
  name: 'Wheelchair Ramp',
  value: '13',
}];
