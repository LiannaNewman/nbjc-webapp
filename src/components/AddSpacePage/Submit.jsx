import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import Button from '@material-ui/core/Button';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Typography from '@material-ui/core/Typography';
import EditIcon from '@material-ui/icons/Edit';

import BusinessResultCard from '../BusinessResultCard';
import ChipList from '../ChipList';
import StarRating from '../StarRating';

import { addSpaceProps as addSpacePropTypes } from '../../types';

const styles = (theme) => ({
  pageTitle: {
    [theme.breakpoints.up('xs')]: {
      marginBottom: 40,
    },
    [theme.breakpoints.up('mobile')]: {
      marginBottom: 60,
    },
  },
  section: {
    marginTop: 20,
  },
  sectionTitle: {
    display: 'inline-block',
    marginBottom: 10,
  },
  edit: {
    display: 'inline-block',
    cursor: 'pointer',
    float: 'right',
  },
  checkbox: {
    maginTop: 20,
    marginBottom: 20,
  },
  footer: {
    margin: '40px 0',
  },
  submitButton: {
    float: 'right',
    marginBottom: 20,
  },
});

const Submit = ({
  classes,
  onBack,
  onSubmit,
  addSpaceProps,
}) => {
  const matches = useMediaQuery('(min-width:376px)');
  const [contact, setContact] = useState(false);
  return (
    <>
      <Typography variant={matches ? 'h4' : 'subtitle1'} align="center" className={classes.pageTitle}>
        Review your submission. You can edit the spaceâ€™s details before
        submitting.
      </Typography>
      <div className={classes.section}>
        <div className={classes.sectionHeader}>
          <Typography
            variant="h6"
            align="center"
            className={classes.sectionTitle}
          >
            Space Details
          </Typography>
          <EditIcon
            color="primary"
            className={classes.edit}
            onClick={() => onBack(1)}
          />
        </div>
        <BusinessResultCard
          business={addSpaceProps.business}
          checked={false}
          hideCheck
        />
      </div>
      <div className={classes.section}>
        <div className={classes.sectionHeader}>
          <Typography
            variant="h6"
            align="center"
            className={classes.sectionTitle}
          >
            Space Attributes
          </Typography>
          <EditIcon
            color="primary"
            className={classes.edit}
            onClick={() => onBack(1)}
          />
        </div>
        <ChipList chips={addSpaceProps.chips} />
      </div>
      <div className={classes.section}>
        <div className={classes.sectionHeader}>
          <Typography
            variant="h6"
            align="center"
            className={classes.sectionTitle}
          >
            Your Rating
          </Typography>
          <EditIcon
            color="primary"
            className={classes.edit}
            onClick={() => onBack(1)}
          />
        </div>
        <StarRating numberFilled={addSpaceProps.rating} editable={false} />
      </div>
      <FormControlLabel
        className={classes.checkbox}
        control={<Checkbox name="anon" color="primary" onChange={() => setContact(!contact)} checked={contact} />}
        label={(
          <Typography variant="body2">
            A National Black Justice Coalition administrator can contact me at
            user@email.com about my submission.
          </Typography>
        )}
      />
      <div className={classes.footer}>
        <Button
          type="submit"
          variant="contained"
          color="secondary"
          className={classes.submitButton}
          fullWidth={!matches}
          onClick={() => onSubmit({ ...addSpaceProps, canContact: contact })}
          disableElevation
        >
          Submit
        </Button>
        <Button
          type="cancel"
          variant="outlined"
          color="secondary"
          fullWidth={!matches}
          onClick={onBack}
          disableElevation
        >
          Back
        </Button>
      </div>
    </>
  );
};

Submit.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  onBack: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  addSpaceProps: PropTypes.shape(addSpacePropTypes).isRequired,
};

Submit.defaultProps = {};

export default withStyles(styles)(Submit);
