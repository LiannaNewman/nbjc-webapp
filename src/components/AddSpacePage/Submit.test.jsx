import React from 'react';
import { render } from '@testing-library/react';

import getYelpResultMock from '../../__mocks__/getYelpResultMock';
import chips from '../../api/chips';

import Submit from './Submit';

jest.mock('react-intersection-observer', () => ({
  useInView: () => [{}, true],
}));

describe('Review', () => {
  test('renders with default values', () => {
    const addSpaceProps = {
      business: getYelpResultMock(),
      chips: [
        { ...chips[0], isSelected: true },
        { ...chips[1], isSelected: true },
        { ...chips[2], isSelected: true },
      ],
      rating: 4,
    };
    const { asFragment } = render(
      <Submit onSubmit={jest.fn()} onBack={jest.fn()} addSpaceProps={addSpaceProps} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
