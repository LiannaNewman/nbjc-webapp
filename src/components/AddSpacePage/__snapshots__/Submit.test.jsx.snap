// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review renders with default values 1`] = `
<DocumentFragment>
  <h6
    class="MuiTypography-root Submit-pageTitle-1 MuiTypography-subtitle1 MuiTypography-alignCenter"
  >
    Review your submission. You can edit the spaceâ€™s details before submitting.
  </h6>
  <div
    class="Submit-section-2"
  >
    <div>
      <h6
        class="MuiTypography-root Submit-sectionTitle-3 MuiTypography-h6 MuiTypography-alignCenter"
      >
        Space Details
      </h6>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root Submit-edit-4 MuiSvgIcon-colorPrimary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
    </div>
    <div
      class=""
      data-testid="businessresultcard-123"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-8 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-10 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-12"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-12"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Submit-section-2"
  >
    <div>
      <h6
        class="MuiTypography-root Submit-sectionTitle-3 MuiTypography-h6 MuiTypography-alignCenter"
      >
        Space Attributes
      </h6>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root Submit-edit-4 MuiSvgIcon-colorPrimary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
    </div>
    <div
      class="ChipList-chipWrapper-13"
    >
      <div
        class="MuiChip-root ChipList-chip-14 MuiChip-colorSecondary"
      >
        <span
          class="MuiChip-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Accessible
          </p>
        </span>
      </div>
      <div
        class="MuiChip-root ChipList-chip-14 MuiChip-colorSecondary"
      >
        <span
          class="MuiChip-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            ATM Available
          </p>
        </span>
      </div>
      <div
        class="MuiChip-root ChipList-chip-14 MuiChip-colorSecondary"
      >
        <span
          class="MuiChip-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            ASL Spoken
          </p>
        </span>
      </div>
    </div>
  </div>
  <div
    class="Submit-section-2"
  >
    <div>
      <h6
        class="MuiTypography-root Submit-sectionTitle-3 MuiTypography-h6 MuiTypography-alignCenter"
      >
        Your Rating
      </h6>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root Submit-edit-4 MuiSvgIcon-colorPrimary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
    </div>
    <button
      class="MuiButtonBase-root MuiIconButton-root StarRating-button-15 StarRating-noHover-16 MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root StarRating-button-15 StarRating-noHover-16 MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root StarRating-button-15 StarRating-noHover-16 MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root StarRating-button-15 StarRating-noHover-16 MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root StarRating-button-15 StarRating-noHover-16 MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
        </svg>
      </span>
    </button>
  </div>
  <label
    class="MuiFormControlLabel-root Submit-checkbox-5"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-20"
          data-indeterminate="false"
          name="anon"
          type="checkbox"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        A National Black Justice Coalition administrator can contact me at user@email.com about my submission.
      </p>
    </span>
  </label>
  <div
    class="Submit-footer-6"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained Submit-submitButton-7 MuiButton-containedSecondary MuiButton-disableElevation MuiButton-fullWidth"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Submit
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-disableElevation MuiButton-fullWidth"
      tabindex="0"
      type="cancel"
    >
      <span
        class="MuiButton-label"
      >
        Back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
